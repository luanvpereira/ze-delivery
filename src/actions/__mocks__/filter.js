export const setCurrentMethod = jest.fn()
	.mockImplementation(() => ({ type: Math.floor(Math.random() * 20) }));

export const changeCategory = jest.fn()
	.mockImplementation(() => ({ type: Math.floor(Math.random() * 20) }));

export const changeFilterText = jest.fn()
	.mockImplementation(() => ({ type: Math.floor(Math.random() * 20) }));
