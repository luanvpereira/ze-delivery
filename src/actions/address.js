import axios from 'axios';

import { GEOCODE_URL, GOOGLE_CREDENTIAL } from '../constants';

export const getAddressess = address => async () => {
	const {
		data: { results: addresses }
	} = await axios.get(`${GEOCODE_URL}?address=${address}&key=${GOOGLE_CREDENTIAL}`);

	return addresses;
};
