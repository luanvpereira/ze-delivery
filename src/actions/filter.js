import {
	CHANGE_METHOD_ADDRESS,
	CHANGE_CURRENT_CATEGORY,
	CHANGE_FILTER_TEXT
} from '../types';

export const setCurrentMethod = payload => ({
	type: CHANGE_METHOD_ADDRESS,
	payload
});

export const changeCategory = payload => ({
	type: CHANGE_CURRENT_CATEGORY,
	payload: parseInt(payload, 10)
});

export const changeFilterText = payload => ({
	type: CHANGE_FILTER_TEXT,
	payload
});
