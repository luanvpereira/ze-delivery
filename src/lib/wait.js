const ids = {};

const wait = (id, time) => {
	return new Promise(resolve => {
		clearTimeout(ids[id]);
		ids[id] = setTimeout(() => {
			resolve();
		}, time);
	});
};

export default wait;
