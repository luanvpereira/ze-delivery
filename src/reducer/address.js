import { combineReducers } from 'redux';

import { CHANGE_CURRENT_ADDRESS } from '../types';

export const currentAddress = (state = null, { type, payload }) => {
	switch (type) {
		case CHANGE_CURRENT_ADDRESS:
			return payload;

		default:
			return state;
	}
};

export default combineReducers({
	currentAddress
});
