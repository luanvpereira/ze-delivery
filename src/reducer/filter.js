import { combineReducers } from 'redux';

import {
	CHANGE_METHOD_ADDRESS,
	CHANGE_CURRENT_CATEGORY,
	CHANGE_FILTER_TEXT
} from '../types';

export const id = (state = null, { type, payload }) => {
	switch (type) {
		case CHANGE_METHOD_ADDRESS:
			return payload;

		default:
			return state;
	}
};

export const categoryId = (state = 0, { type, payload }) => {
	switch (type) {
		case CHANGE_CURRENT_CATEGORY:
			return payload;

		default:
			return state;
	}
};

export const search = (state = '', { type, payload }) => {
	switch (type) {
		case CHANGE_FILTER_TEXT:
			return payload;

		default:
			return state;
	}
};

export default combineReducers({
	id,
	categoryId,
	search
});
