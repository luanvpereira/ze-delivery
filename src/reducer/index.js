import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

export const middlewares = () => [thunk];

export const initializeStore = (initialState = {}) =>
	createStore(
		combineReducers({}),
		initialState,
		composeWithDevTools(applyMiddleware(...middlewares())),
	);
