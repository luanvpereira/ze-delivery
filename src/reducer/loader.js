import {
	SHOW_LOADER,
	HIDE_LOADER
} from '../types';

export default (state = false, { type }) => {
	switch (type) {
		case SHOW_LOADER:
			return true;

		case HIDE_LOADER:
			return false;

		default:
			return state;
	}
};
