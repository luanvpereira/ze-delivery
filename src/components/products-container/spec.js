jest.mock('../../actions/filter');

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from '../../lib/wait';
import { initializeStore } from '../../reducer';
import * as filterActions from '../../actions/filter';

import ProductsContainer, {
	ALL_CATEGORIES_QUERY,
	POC_CATEGORY_SEARCH_QUERY
} from './';

const defaultStoreValues = {
	filter: {
		id: '243',
		search: '',
		categoryId: 0
	}
};

const mocks = [
	{
		request: {
			query: ALL_CATEGORIES_QUERY
		},
		result: {
			data: {
				allCategory: [
					{
						title: 'Balas & Doces',
						id: '1096'
					},
					{
						title: 'Cervejas',
						id: '1098'
					},
					{
						title: 'Cervejas Especiais & Importadas',
						id: '1099'
					},
					{
						title: 'Cervejas RetornÃ¡veis.',
						id: '1101'
					}
				]
			}
		}
	},
	{
		request: {
			query: POC_CATEGORY_SEARCH_QUERY,
			variables: {
				id: '243',
				search: '',
				categoryId: 0
			}
		},
		result: {
			data: {
				poc: {
					products: [
						{
							productVariants: [
								{
									title: 'Stella Artois 275ml - Unidade',
									description: null,
									imageUrl: 'https://s3-us-west-2.amazonaws.com/ze-delivery/upload/images/592de3ed8977d.jpg',
									price: 3.69
								}
							]
						},
						{
							productVariants: [
								{
									title: 'Dieguinho',
									description: 'teste',
									imageUrl: 'https://s3-us-west-2.amazonaws.com/courier-images-dev/product/00001545_22e9df99-9e2e-49a6-af9a-4dd187dde8f4.jpg',
									price: null
								}
							],
						},
						{
							productVariants: [
								{
									title: 'Dieguinho',
									description: 'Dieguinho',
									imageUrl: 'https://s3-us-west-2.amazonaws.com/courier-images-dev/product/00001546_6bd5491d-abcf-4245-9668-789e27964827.jpg',
									price: null
								}
							]
						},
						{
							productVariants: [
								{
									title: 'Chelsea',
									description: 'Chelsea',
									imageUrl: 'https://s3-us-west-2.amazonaws.com/courier-images-dev/product/00001549_668d4366-f34a-40c9-8ba0-62a3b5da56d8.jpg',
									price: null
								}
							]
						}
					]
				}
			}
		}
	}
];

const getWrapper = ({
	render = mount,
	props = {},
	store = defaultStoreValues
} = {}) => render(
	<MockedProvider mocks={mocks} addTypename={false}>
		<Provider store={initializeStore(store)}>
			<ProductsContainer {...props} />
		</Provider>
	</MockedProvider>
);

describe('<ProductContainer/>', () => {
	describe('#handleCategoryChange', () => {
		it('should call #changeCategory with value passed event parameter', () => {
			const wrapper = getWrapper();

			const event = {
				target: {
					value: 1
				}
			};

			wrapper.find('ProductsContainer').instance().handleCategoryChange(event);

			expect(filterActions.changeCategory).toHaveBeenCalledWith(event.target.value);
		});
	});

	describe('#handleSearchChange', () => {
		it('should call #changeFilterText with value passed event parameter', () => {
			const wrapper = getWrapper();

			const event = {
				target: {
					value: 'Breja'
				}
			};

			wrapper.find('ProductsContainer').instance().handleSearchChange(event);

			expect(filterActions.changeFilterText).toHaveBeenCalledWith(event.target.value);
		});
	});

	describe('#render', () => {
		it('should render a <ProductsContainer/>', async () => {
			const wrapper = getWrapper();

			await wait(0);

			wrapper.update();

			expect(wrapper.find('ProductsContainer')).toMatchSnapshot();
		});

		it('should render a <ProductsContainer/> without graphql states', () => {
			const wrapper = getWrapper();
			expect(wrapper.find('ProductsContainer')).toMatchSnapshot();
		});
	});
});
