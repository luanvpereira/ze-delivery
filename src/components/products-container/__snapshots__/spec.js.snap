// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductContainer/> #render should render a <ProductsContainer/> 1`] = `
<ProductsContainer
  categories={
    Object {
      "allCategory": Array [
        Object {
          "id": "1096",
          "title": "Balas & Doces",
        },
        Object {
          "id": "1098",
          "title": "Cervejas",
        },
        Object {
          "id": "1099",
          "title": "Cervejas Especiais & Importadas",
        },
        Object {
          "id": "1101",
          "title": "Cervejas Retornáveis.",
        },
      ],
      "error": undefined,
      "fetchMore": [Function],
      "loading": false,
      "networkStatus": 7,
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {},
    }
  }
  changeCategory={[Function]}
  changeFilterText={[Function]}
  filter={
    Object {
      "categoryId": 0,
      "id": "243",
      "search": "",
    }
  }
  setCurrentMethod={[Function]}
>
  <div>
    <select
      onChange={[Function]}
    >
      <option
        value="0"
      >
        Todos
      </option>
      <option
        key="1096"
        value="1096"
      >
        Balas & Doces
      </option>
      <option
        key="1098"
        value="1098"
      >
        Cervejas
      </option>
      <option
        key="1099"
        value="1099"
      >
        Cervejas Especiais & Importadas
      </option>
      <option
        key="1101"
        value="1101"
      >
        Cervejas Retornáveis.
      </option>
    </select>
    <input
      defaultValue=""
      onChange={[Function]}
      type="text"
    />
    <Query
      query={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "pocCategorySearch",
              },
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                        },
                      },
                    ],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "poc",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [
                            Object {
                              "kind": "Argument",
                              "name": Object {
                                "kind": "Name",
                                "value": "categoryId",
                              },
                              "value": Object {
                                "kind": "Variable",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "categoryId",
                                },
                              },
                            },
                            Object {
                              "kind": "Argument",
                              "name": Object {
                                "kind": "Name",
                                "value": "search",
                              },
                              "value": Object {
                                "kind": "Variable",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "search",
                                },
                              },
                            },
                          ],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "products",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "productVariants",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "title",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "description",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "imageUrl",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "price",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [
                Object {
                  "defaultValue": undefined,
                  "directives": Array [],
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "ID",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "id",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "directives": Array [],
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "search",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "directives": Array [],
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "Int",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "categoryId",
                    },
                  },
                },
              ],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 240,
            "start": 0,
          },
        }
      }
      variables={
        Object {
          "categoryId": 0,
          "id": "243",
          "search": "",
        }
      }
    >
      <ul>
        <li
          key="0"
        >
          <strong>
            Stella Artois 275ml - Unidade
          </strong>
           
          <br />
          <img
            src="https://s3-us-west-2.amazonaws.com/ze-delivery/upload/images/592de3ed8977d.jpg"
          />
          <br />
          <strong>
            3.69
          </strong>
          <br />
        </li>
        <li
          key="1"
        >
          <strong>
            Dieguinho
          </strong>
           
          <br />
          <img
            src="https://s3-us-west-2.amazonaws.com/courier-images-dev/product/00001545_22e9df99-9e2e-49a6-af9a-4dd187dde8f4.jpg"
          />
          <br />
          <strong />
          <br />
        </li>
        <li
          key="2"
        >
          <strong>
            Dieguinho
          </strong>
           
          <br />
          <img
            src="https://s3-us-west-2.amazonaws.com/courier-images-dev/product/00001546_6bd5491d-abcf-4245-9668-789e27964827.jpg"
          />
          <br />
          <strong />
          <br />
        </li>
        <li
          key="3"
        >
          <strong>
            Chelsea
          </strong>
           
          <br />
          <img
            src="https://s3-us-west-2.amazonaws.com/courier-images-dev/product/00001549_668d4366-f34a-40c9-8ba0-62a3b5da56d8.jpg"
          />
          <br />
          <strong />
          <br />
        </li>
      </ul>
    </Query>
  </div>
</ProductsContainer>
`;

exports[`<ProductContainer/> #render should render a <ProductsContainer/> without graphql states 1`] = `
<ProductsContainer
  categories={
    Object {
      "error": undefined,
      "fetchMore": [Function],
      "loading": true,
      "networkStatus": 1,
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {},
    }
  }
  changeCategory={[Function]}
  changeFilterText={[Function]}
  filter={
    Object {
      "categoryId": 0,
      "id": "243",
      "search": "",
    }
  }
  setCurrentMethod={[Function]}
>
  <div>
    <select
      onChange={[Function]}
    >
      <option
        value="0"
      >
        Todos
      </option>
    </select>
    <input
      defaultValue=""
      onChange={[Function]}
      type="text"
    />
    <Query
      query={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "pocCategorySearch",
              },
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                        },
                      },
                    ],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "poc",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [
                            Object {
                              "kind": "Argument",
                              "name": Object {
                                "kind": "Name",
                                "value": "categoryId",
                              },
                              "value": Object {
                                "kind": "Variable",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "categoryId",
                                },
                              },
                            },
                            Object {
                              "kind": "Argument",
                              "name": Object {
                                "kind": "Name",
                                "value": "search",
                              },
                              "value": Object {
                                "kind": "Variable",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "search",
                                },
                              },
                            },
                          ],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "products",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "productVariants",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "title",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "description",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "imageUrl",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "price",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [
                Object {
                  "defaultValue": undefined,
                  "directives": Array [],
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "ID",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "id",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "directives": Array [],
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "search",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "directives": Array [],
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "Int",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "categoryId",
                    },
                  },
                },
              ],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 240,
            "start": 0,
          },
        }
      }
      variables={
        Object {
          "categoryId": 0,
          "id": "243",
          "search": "",
        }
      }
    >
      <ul />
    </Query>
  </div>
</ProductsContainer>
`;
